app/workers/stale_issue_marker.rb

```diff
@@ -23,49 +23,50 @@ class StaleIssueMarker
 Thank you for all your contributions!
   EOS
 
-  attr_reader :fq_repo_name
-
-  def perform(fq_repo_name)
-    @fq_repo_name = fq_repo_name
-    raise "The label #{STALE_LABEL_NAME} does not exist on #{fq_repo_name}" unless GithubService.valid_label?(fq_repo_name, STALE_LABEL_NAME)
-
-    GithubService.issues(fq_repo_name, :state => :open, :sort => :updated, :direction => :asc).each do |issue|
-      pinned = (issue.labels & PINNED_LABELS).present?
-
-      if issue.updated_at < stale_date && !pinned
-        if issue.pull_request?
-          closable_prs << issue
-        else
-          stale_issues << issue
-        end
+  def perform
+    GithubService.search_issues(*search_args).each do |issue|
+      if issue.pull_request?
+        close_pr(issue)
+      else
+        mark_as_stale(issue)
       end
     end
-
-    stale_issues.each do |issue|
-      next if issue.labels.include?(STALE_LABEL_NAME)
-      logger.info("[#{Time.now.utc}] - Marking issue #{fq_repo_name}##{issue.number} as stale")
-      issue.add_labels([STALE_LABEL_NAME])
-      issue.add_comment(STALE_ISSUE_MESSAGE)
-    end
-
-    closable_prs.each do |pr|
-      logger.info("[#{Time.now.utc}] - Closing stale PR #{fq_repo_name}##{pr.number}")
-      GithubService.close_pull_request(pr.fq_repo_name, pr.number)
-      pr.add_comment(CLOSABLE_PR_MESSAGE)
-    end
   end
 
   private
 
+  def search_args
+    query  = "is:open archived:false update:<#{stale_date.strftime('%Y-%m-%d')}"
```

Comment:
what  does `update:<date` entail?  Like, if someone pushes a commit, does that count? or perhaps adds a label or assignee?

Reply:
It is the query equivalent of:

`if issue.updated_at < stale_date && !pinned`

With the `!pinned` part happening below.

Regarding the dates specifically, more info can be found here:

https://help.github.com/en/github/searching-for-information-on-github/searching-issues-and-pull-requests#search-by-when-an-issue-or-pull-request-was-created-or-last-updated
https://help.github.com/en/github/searching-for-information-on-github/understanding-the-search-syntax#query-for-dates

For the example for `<` with dates is:

> Query | Example
> -- | --
> `<YYYY-MM-DD` | `cats pushed:<2012-07-05`  matches code with the word "cats" in repositories that were pushed to before July 5, 2012.
```

So translating to `update` should mean that anything with the most recent update happening before `6.mounts.ago`.

Reply:
I'm sorry...I worded my question poorly.  I'm more asking if checking "update" is sufficient for finding all of the things people can do with a PR.  For example if someone pushes a commit is "update" updated?  I'm concerned that we mark something stale even if someone does something to the PR because the search query is not aware of some specific kind of interaction.

Reply:
@Fryguy nah, I don't think you worded it poorly, I was trying to answer both

>  I'm more asking if checking "update" is sufficient for finding all of the things people can do with a PR.

The [first link](https://help.github.com/en/github/searching-for-information-on-github/searching-issues-and-pull-requests#search-by-when-an-issue-or-pull-request-was-created-or-last-updated) above is the best I can do to answer that, where it says:

> You can filter issues based on times of creation, or when they were last updated. For issue creation, you can use the created qualifier; to find out when an issue was last updated, you'll want to use the updated qualifier.

But if that is as you described, I can only say "I think so?" at best... seems like it would be, but I agree with you in that I am slightly concerned if this is an equivalent.

* * *

I guess the best I could do is a test.  So starting with this:

[`org:ManageIQ label:stale updated:<2019-02-28 sort:updated-desc`](https://github.com/pulls?utf8=%E2%9C%93&q=org%3AManageIQ+label%3Astale+updated%3A%3C2019-02-28+sort%3Aupdated-desc)

I get this as the first result:  https://github.com/ManageIQ/manageiq-providers-redfish/issues/22

If I bump back the date to `2019-02-24`:

[`org:ManageIQ label:stale updated:<2019-02-24 sort:updated-desc`](https://github.com/pulls?utf8=✓&q=org%3AManageIQ+label%3Astale+updated%3A<2019-02-24+sort%3Aupdated-desc)

It is no longer found on the list.

That said, I don't see that as a good example, since we are dealing with an Issue that doesn't deal with commit events, so I decided to try it with `is:pr` just to limit it a bit more to "things with commits":

[`is:pr org:ManageIQ label:stale updated:<2019-02-28 sort:updated-desc`](https://github.com/pulls?utf8=✓&q=is%3Apr+org%3AManageIQ+label%3Astale+updated%3A<2019-02-28+sort%3Aupdated-desc)

And this was the PR that I got at the top:  https://github.com/ManageIQ/manageiq/pull/3870

(it is funny to see the `@oleg-bot` at work on that one 😄 )

That PR has a "updated on Nov 28, 2017" in the result, so if we updated the date to the `2017-11-29`:

[`is:pr org:ManageIQ label:stale updated:<2017-11-29 sort:updated-desc`](https://github.com/pulls?utf8=%E2%9C%93&q=is%3Apr+org%3AManageIQ+label%3Astale+updated%3A%3C2017-11-29+sort%3Aupdated-desc)

The PR is still in the results, but bumping it back a day:

[`is:pr org:ManageIQ label:stale updated:<2017-11-28 sort:updated-desc`](https://github.com/pulls?utf8=%E2%9C%93&q=is%3Apr+org%3AManageIQ+label%3Astale+updated%3A%3C2017-11-28+sort%3Aupdated-desc)

It is removed, and it looks like what was registered as an updated was a [branch delete action by GregB](https://github.com/ManageIQ/manageiq/pull/3870#event-1361997212), so I think we can be relatively sure that this will cover commits...

I think... check my logic on that break down...

Reply:
This sounds good to me.  Thanks for the analysis.